GeofenceDisplay/src/geofence-display.ts

```diff
@@ -1,155 +1,111 @@
-export {}
-
 import L from 'leaflet'
+// import Gubu from 'gubu'
 import './geofence-display.css'
+// import { RasterCoords } from './rastercoords'
+
+// TODO: implement Gubu with interfaces
+interface Geofence {
+  id: string
+  title: string
+  latlngs: Array<Array<string>>
+  colour: string
+}
 
-const PlantquestGeofenceDisplay = L.Layer.extend({
-  options: {
-    // geofenceLabelPane.style.zIndex = 235
-    // provide geofence data to component using an option
-  },
+interface PlantquestGeofenceDisplayOptions extends L.ControlOptions {
+  debug: boolean
+  geofences: Geofence[]
+}
 
+const PlantquestGeofenceDisplay = L.Layer.extend({
+  // Use GUBU for options validation
   initialize: function (this: any, options: any) {
-    console.log('PGeoD init')
+    let validatedOptions = options as PlantquestGeofenceDisplayOptions
```

Comment:
validatedOptions is a long name and these tend to create "noisy" code
since we are validating, it is safe to rename thus:
```
initialize: function (this: any, rawOptions: PlantquestGeofenceDisplayOptions) {
    // use const PlantquestGeofenceDisplayOptionsShape = Gubu({}) to get started - this will allow anything
    let options = PlantquestGeofenceDisplayOptionsShape(rawOptions)
```